### ISSUE TITLE
too many cash shop item crash

### ISSUE THREAD

**Issue by ChiesaV** [rathena/rathena#7757]
### rAthena Hash

latest

### Client Date

2022-04-06 Kro from Nemo with RSU, Ragnarok English

### Server Mode

Renewal

### Result

[Error]: _mmalloc: 4294950912
[Fatal Error]: Received a crash signal, trying to handle it as good as possible!
[Error]: Server received crash signal! Attempting to save all online characters!
on map server
When too many cash shop item, but it dont happend in the previous versin of rathena (Ep 18, item_cash still using txt).

### Expected Result

Not crash, because old rathena which use item cash txt is capable of it.

### How to Reproduce

i can provide the file.

### Official Information

2022 04 kro one from nemo.

**Comment by aleos89** [comment]
The doubt sets in with you using the latest version of rAthena when you state your item_cash is still in TXT format. I'm not sure what the client limit is but maybe someone else knows. How many items do you have set in your database?

**Comment by ChiesaV** [comment]
No, I mean, it was fine during rathena item_cash txt version.
I have 2 version of rathena and ragnarok.
It upto 16500 item in the txt, and almost the same in the yml version.
But the yml version one is crashed directly after login.

**Comment by vstumpf** [comment]
I don't see a reason why the cashshopdb would allocate that much data.

Are you sure it's not the itemdb that is causing the crash?
Is it _only_ the cashdb changing from txt to yml? Can you run the server in a debugger so we get a full stacktrace?

**Comment by ChiesaV** [comment]
hey vstumpf, 
could you give me link / brief how to do it ?
and for the conclusion i came up with, because I try to reduce it with 3 different items randomly. and it always succeed when the data amount is on 25% of the total I initially try.
and why I submit this with bug tag, because the txt version is capable of doing it. Even though I understand that it for cheating and personal gameplay purpose to have that many of database. 
It would be great if we can have same capability with the old one tho..

**Comment by ChiesaV** [comment]
when I input Infinite_Concentration_Potion into the yml, it said not found and not displayed at cash shop,
but command gm using itemid 12884 it is work, using @item Infinite_Concentration_Potion not work.
Meanwhile, using while using infinite_berserk_potion, both itemid and aegisname work at using command gm @item.

**Comment by ChiesaV** [comment]
seems the root cause is, every tab, can't have more than around 250-260 pages. if more, crash after character selection.

**Comment by Lemongrass3110** [comment]
@ChiesaV can you please add #7763 to your server and tell us which file tries to allocate 4GB of RAM?

**Comment by ChiesaV** [comment]
will try. and inform you later.

**Comment by ChiesaV** [comment]
[Error]: _mmalloc: Invalid allocation size 4294950912 bytes at F:\RagnarokOffline\Server\rathena-master\src\common\socket.cpp:791
[Fatal Error]: Received a crash signal, trying to handle it as good as possible!
[Error]: Server received crash signal! Attempting to save all online characters!

**Comment by Lemongrass3110** [comment]
Thank you!
We are getting closer.
Can you additionally add 251b801 and try again?

**Comment by ChiesaV** [comment]
will do, a moment

**Comment by ChiesaV** [comment]
[Fatal Error]: Received a crash signal, trying to handle it as good as possible!
[Error]: Server received crash signal! Attempting to save all online characters!

**Comment by Lemongrass3110** [comment]
Are you sure that you did it correcly?

**Comment by ChiesaV** [comment]
Yes, it compiled correctly.

**Comment by Lemongrass3110** [comment]
https://raw.githubusercontent.com/rathena/rathena/d447f9ea48ce7e7d533d91d54ae6aa06542609b6/src/common/malloc.cpp
https://raw.githubusercontent.com/rathena/rathena/d447f9ea48ce7e7d533d91d54ae6aa06542609b6/src/common/malloc.hpp
https://raw.githubusercontent.com/rathena/rathena/d447f9ea48ce7e7d533d91d54ae6aa06542609b6/src/common/socket.cpp
https://raw.githubusercontent.com/rathena/rathena/d447f9ea48ce7e7d533d91d54ae6aa06542609b6/src/common/socket.hpp

**Comment by ChiesaV** [comment]
a moment

**Comment by ChiesaV** [comment]
[Fatal Error]: Received a crash signal, trying to handle it as good as possible!
[Error]: Server received crash signal! Attempting to save all online characters!
[Debug]: (pc_makesavestatus) mapindex_id2name: Requested name for non-existant map index [0] in cache.

Tried second time.
[Fatal Error]: Received a crash signal, trying to handle it as good as possible!
[Error]: Server received crash signal! Attempting to save all online characters!

**Comment by Lemongrass3110** [comment]
```
[Info]: 'test' logged in. (AID/CID: '2000000/150001', IP: '127.0.0.1', Group '99').
[Error]: _mmalloc: Invalid allocation size 4000006144 bytes at C:\Users\Lemongrass\Desktop\rathena\src\common\socket.cpp:879
```

Works perfectly fine for me.
Maybe try to run your rAthena in Microsoft Visual Studio in Debug Mode.
If you need help with this, please visit #support in our Discord.

**Comment by ChiesaV** [comment]
is there any solution to this sir @Lemongrass3110 ?

**Comment by Akkarinage** [comment]
> is there any solution to this sir @Lemongrass3110 ?, the main bug

Given that you were asked to follow instructions and haven't provided the requested feedback, I doubt there's new progress based on no additional information.

**Comment by ChiesaV** [comment]
https://drive.google.com/file/d/1va32SE5Vfoz66yAmIcwoskWzo2q0XKTh/view?usp=share_link
here we go
no, i just compiled it from item_db_re

**Comment by ChiesaV** [comment]
I solved it temporarily by
1. turning off cash shop preview feature on src / core then compile it.
2. then using warp, to handle the price / random price error on cash shop
3. capacity per category increase from 250 pages to around 400 pages. still crashed on login  if above 450 pages per category.

**Comment by Lemongrass3110** [comment]
```
[Status]: Loading '8' entries in 'db/import/item_cash.yml'
[Error]: Cash item 0 does not exist, skipping.
[Error]: Occurred in file 'db/import/item_cash.yml' on line 17053 and column 7.
Item: 0
[Status]: Done reading '8' entries in 'db/import/item_cash.yml'
```

But beside that I fixed the real issue in #7763.
Due to ENABLE_CASHSHOP_PREVIEW_PATCH the packet got too big.

**Comment by ChiesaV** [comment]
pulled all your change in the #7763
confirmed it already fixed (I already turned back on the preview patch.
thank you so much, amazing.
